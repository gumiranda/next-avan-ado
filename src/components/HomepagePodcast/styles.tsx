import styled from 'styled-components';

export const Wrapper = styled.div`
  padding: 0 5rem;
  height: calc(82vh - 6.5rem);
  overflow-y: scroll;
  h2 {
    margin-top: 3rem;
    margin-bottom: 1.5rem;
  }
`;
