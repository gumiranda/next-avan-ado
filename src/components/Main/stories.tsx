import Main from './Main';
import { Story, Meta } from '@storybook/react/types-6-0';
export default {
  title: 'Main',
  /* forma alternativa de passar args
  args: { title: 'lalala', description: 'nnnn' },
   */
  component: Main
} as Meta;

export const Basic: Story = (args) => <Main {...args} />;

Basic.args = {
  title: 'React Avan√ßado',
  description: 'Typescript, ReactJS, NextJS e Styled Components'
};
