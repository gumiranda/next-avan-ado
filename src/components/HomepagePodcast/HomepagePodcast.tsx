import { ReactNode } from 'react';
import * as S from './styles';
interface IProps {
  children: ReactNode;
}
const HomepagePodcast = ({ children, ...props }: IProps) => (
  <S.Wrapper>{children}</S.Wrapper>
);

export default HomepagePodcast;
