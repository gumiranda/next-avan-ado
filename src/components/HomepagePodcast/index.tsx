import HomepagePodcast from './HomepagePodcast';
export default HomepagePodcast;
