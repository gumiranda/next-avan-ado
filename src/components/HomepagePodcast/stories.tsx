import { Story, Meta } from '@storybook/react/types-6-0';

import HomepagePodcast from './HomepagePodcast';
import Main from 'components/Main';

export default {
  title: 'HomepagePodcast',
  component: HomepagePodcast
} as Meta;

export const Basic: Story = (args) => (
  <HomepagePodcast {...args}>
    <Main />
  </HomepagePodcast>
);
