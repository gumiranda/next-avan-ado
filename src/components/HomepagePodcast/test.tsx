import { render, screen } from '@testing-library/react';
import Main from '../Main';

import HomepagePodcast from './HomepagePodcast';

describe('<HomepagePodcast />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <HomepagePodcast>
        <Main />
      </HomepagePodcast>
    );
    expect(screen.getByRole('heading', { name: /react avan√ßado/i })).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
