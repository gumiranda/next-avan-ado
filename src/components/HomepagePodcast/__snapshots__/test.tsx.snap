// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomepagePodcast /> should render the heading 1`] = `
.c1 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 4rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  text-align: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c3 {
  font-size: 2.5rem;
}

.c4 {
  font-size: 2rem;
  font-weight: 400;
}

.c5 {
  margin-top: 3rem;
  width: min(30rem,100%);
}

.c0 {
  padding: 0 5rem;
  height: calc(82vh - 6.5rem);
  overflow-y: scroll;
}

.c0 h2 {
  margin-top: 3rem;
  margin-bottom: 1.5rem;
}

<div
  class="c0"
>
  <main
    class="c1"
  >
    <img
      alt="Image de um átomo e React Avançado escrito ao lado."
      class="c2"
      src="/img/logo.svg"
    />
    <h1
      class="c3"
    >
      React Avançado
    </h1>
    <h2
      class="c4"
    >
      Typescript, ReactJS, NextJS e Styled Components
    </h2>
    <img
      alt="Um desenvolvedor de frente para uma tela com código"
      class="c5"
      src="/img/hero-illustration.svg"
    />
  </main>
</div>
`;
